apps/system/utils/fscmd.c

```diff
@@ -1357,20 +1378,36 @@ static int format_filesystem(fs_minor_t minor)
 		printf("Invalid minor number : %d", minor);
 		return ERROR;
 	}
-	for (int i = 0; i < 9; i++) {
-		snprintf(name, sizeof(name), "/dev/smart%dp%d", minor, i);
-		fd = open(name, O_RDWR);
-		if (fd < 0) {
-			continue;
-		}
-		/* TODO Multi root of smartfs should be considered when it enabled */
-		ret = ioctl(fd, BIOC_BULKERASE, 0);
-		close(fd);
-		if (ret != OK) {
-			printf("Low level format failed ret : %d errno : %d", ret, errno);
-			return ret;
-		}
-		break;
+	int index = ERROR;
+	if (minor == FS_BLOCK_MINOR_SECONDARY) {
+#ifdef CONFIG_SECOND_FLASH_PART_NAME
+			index = get_partition_index(CONFIG_SECOND_FLASH_PART_NAME, USERFS_PART_NAME);
+#else
+			printf("config CONFIG_SECOND_FLASH_PARTITION_NAME must be enabled to format\n");
+			return ERROR;
+#endif
+	} else {
+#ifdef CONFIG_FLASH_PART_NAME
+		index = get_partition_index(CONFIG_FLASH_PART_NAME, USERFS_PART_NAME);
```

Comment:
TJ's concern here is that using a kernel config (flash partition list) to complete a fs related TASH command is not desirable.
However, the current implementation loops over all partitions and tries to format them.
One additional problem is that the loop is limited to 10 while in some cases the userfs partition's index might be greater than 9.
So we want to find a middle ground for this, what is your idea @swasthik22